core/chains/evm/gas/fee_history_estimator.go

```diff
@@ -0,0 +1,414 @@
+		return
+	}
+
+	bumpedMaxPriorityFeePerGas := originalFee.TipCap.AddPercentage(u.config.BumpPercent)
+	bumpedMaxFeePerGas := originalFee.FeeCap.AddPercentage(u.config.BumpPercent)
+
+	if u.config.HasMempool {
+		bumpedMaxPriorityFeePerGas, err = LimitBumpedFee(originalFee.TipCap, currentDynamicPrice.TipCap, bumpedMaxPriorityFeePerGas, maxPrice)
+		if err != nil {
+			return bumped, fmt.Errorf("maxPriorityFeePerGas error: %s", err.Error())
+		}
+
+		priorityFeeThreshold, e := u.getPriorityFeeThreshold()
+		if e != nil {
+			err = e
+			return
+		}
+
+		// If either of these two values are 0 it could be that the network has extremely low priority fees. We should skip the
+		// connectivity check because we're only going to be charged for the base fee, which is mandatory.
+		if (priorityFeeThreshold.Cmp(assets.NewWeiI(0)) > 0) && (bumpedMaxPriorityFeePerGas.Cmp(assets.NewWeiI(0)) > 0) &&
+			bumpedMaxPriorityFeePerGas.Cmp(priorityFeeThreshold) > 0 {
+			return bumped, fmt.Errorf("bumpedMaxPriorityFeePergas: %s is above market's %sth percentile: %s, bumping is halted",
+				bumpedMaxPriorityFeePerGas, strconv.Itoa(ConnectivityPercentile), priorityFeeThreshold)
+		}
+	} else {
+		// If the network doesn't have a mempool then transactions are processed in a FCFS manner and maxPriorityFeePerGas value is irrelevant.
+		// We just need to cap the value at maxPrice in case maxFeePerGas also gets capped.
+		bumpedMaxPriorityFeePerGas = assets.WeiMin(bumpedMaxPriorityFeePerGas, maxPrice)
+	}
+
+	bumpedMaxFeePerGas, err = LimitBumpedFee(originalFee.FeeCap, currentDynamicPrice.FeeCap, bumpedMaxFeePerGas, maxPrice)
+	if err != nil {
+		return bumped, fmt.Errorf("maxFeePerGas error: %s", err.Error())
+	}
+
+	bumpedFee := DynamicFee{FeeCap: bumpedMaxFeePerGas, TipCap: bumpedMaxPriorityFeePerGas}
+	u.logger.Debugw("bumped dynamic fee", "originalFee", originalFee, "bumpedFee", bumpedFee)
+
+	return bumpedFee, nil
+}
+
+// LimitBumpedFee selects the maximum value between the bumped attempt and the current fee, if there is one. If the result is higher than the max price it gets capped.
+// Geth's implementation has a hard 10% minimum limit for the bumped values, otherwise it rejects the transaction with an error.
+// See: https://github.com/ethereum/go-ethereum/blob/bff330335b94af3643ac2fb809793f77de3069d4/core/tx_list.go#L298
+//
+// Note: for chains that support EIP-1559 but we still choose to send Legacy transactions to them, the limit is still enforcable due to the fact that Legacy transactions
+// are treated the same way as Dynamic transactions under the hood. For chains that don't support EIP-1559 at all, the limit isn't enforcable but a 10% minimum bump percentage
+// makes sense anyway.
+func LimitBumpedFee(originalFee *assets.Wei, currentFee *assets.Wei, bumpedFee *assets.Wei, maxPrice *assets.Wei) (*assets.Wei, error) {
+	if currentFee != nil {
+		bumpedFee = assets.WeiMax(currentFee, bumpedFee)
+	}
+	bumpedFee = assets.WeiMin(bumpedFee, maxPrice)
```

Comment:
If the bump fee gets capped here, should we check that difference between the original and the capped fee is still at least the `MinimumBumpPercentage`?

Reply:
We do it [here](https://github.com/smartcontractkit/chainlink/pull/13833/files#diff-31090b44bf40193bf55a03b3422e405a89041337bdc9d7fb6ddd5bfc277840feR395)

Reply:
Oh duh sorry that was obvious. Confused myself with the scenario in the comment